### PR TITLE
Remove gogoproto options from gatewayserver.proto and join.proto

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Remove gogoproto options from gatewayserver.proto and join.proto

#### Changes

- Remove gogoproto options from gatewayserver.proto and join.proto
- Fix usage and run tests

#### Testing

Existing tests should cover this.

##### Regressions

Don't expect any.

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/networkserver/networkserver_util_internal_test.go [hunk 1]
```diff
@@ -1200,7 +1200,9 @@ func (env TestEnvironment) AssertScheduleJoinAccept(ctx context.Context, dev *tt
 					events.WithIdentifiers(&dev.EndDeviceIdentifiers),
 				).New(ctx),
 				EvtScheduleJoinAcceptSuccess.With(
-					events.WithData(&ttnpb.ScheduleDownlinkResponse{}),
+					events.WithData(&ttnpb.ScheduleDownlinkResponse{
+						Delay: ttnpb.ProtoDurationPtr(0),
+					}),
```
Comment (KrishnaIyer) [inline pkg/networkserver/networkserver_util_internal_test.go:1205]: For successful downlinks, this cannot be empty. So we get a zero value here and below.

File: pkg/joinserver/http_interop.go [hunk 2]
```diff
@@ -105,16 +105,20 @@ func (srv interopServer) JoinRequest(ctx context.Context, in *interop.JoinReq) (
...
 		},
-		Lifetime:     uint32(res.Lifetime / time.Second),
-		AppSKey:      (*interop.KeyEnvelope)(res.AppSKey),
-		SessionKeyID: interop.Buffer(res.SessionKeyId),
+		AppSKey:      (*interop.KeyEnvelope)(res.SessionKeys.AppSKey),
+		SessionKeyID: interop.Buffer(res.SessionKeys.SessionKeyId),
 	}
+
+	if res.Lifetime != nil {
+		ans.Lifetime = uint32(ttnpb.StdDurationOrZero(res.Lifetime) / time.Second)
```
Comment (adriansmares) [inline pkg/joinserver/http_interop.go]: Why the `nil` check ? 0 divided by something is still 0, right ?

File: pkg/networkserver/grpc_gsns.go [hunk 3]
```diff
@@ -1135,9 +1136,13 @@ func (ns *NetworkServer) handleJoinRequest(ctx context.Context, up *ttnpb.Uplink
 		RawPayload:         up.RawPayload,
 		RxDelay:            macState.DesiredParameters.Rx1Delay,
 		SelectedMacVersion: matched.LorawanVersion, // Assume NS version is always higher than the version of the device
-		DownlinkSettings:   dlSettings,
-		ConsumedAirtime:    up.ConsumedAirtime,
-	})
+		DownlinkSettings:   &dlSettings,
+	}
+	if up.ConsumedAirtime != nil {
+		jReq.ConsumedAirtime = ttnpb.ProtoDuration(up.ConsumedAirtime)
```
Comment (adriansmares) [inline pkg/networkserver/grpc_gsns.go:1142]: Same here - `ProtoDurationOrZero` does this (0 if `nil`).

File: pkg/joinserver/http_interop.go (hunk 2 above)
Comment (KrishnaIyer) [reply pkg/joinserver/http_interop.go:113]: https://github.com/TheThingsNetwork/lorawan-stack/blob/5b38d7e6a5cea6210c88afef918e74bfd11a2f90/pkg/joinserver/joinserver.go#L552-L555
The standard JS implementation does not set this field and it's optional.

Comment (KrishnaIyer) [reply pkg/joinserver/http_interop.go]: Ah yes right

File: pkg/networkserver/grpc_gsns.go (hunk 3 above)
Comment (KrishnaIyer) [reply pkg/networkserver/grpc_gsns.go:1142]: `ProtoDurationOrZero`: There's no such function, but I guess we can add one.

Comment (adriansmares) [reply pkg/networkserver/grpc_gsns.go:1142]: I was wrong - your original code was correct - there is no need for this function.

**Review by adriansmares — APPROVED** [review]

**Review by johanstokking — APPROVED** [review]
LGTM. I'll get some merge conflicts in https://github.com/TheThingsNetwork/lorawan-stack/pull/4917 but I'll fix that when this is merged

**Pull request merged**